ruby-challenges.rb

```diff
@@ -3,25 +3,85 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
 
-beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+# Process
+# Create a method called "letter" that takes in an array and a letter as a parameter
+# create an empty array called "final_array" that will contain our desired values 
+# Use the array.each method to loop through each word in the provided arrays and determine if the words contain the letter
+# If the word contain the letter then the word will be added to our "final_array".
 
-letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-letter_t = 't'
+# beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+
+# letter_o = 'o'
+# # Expected output: ['coffee', 'soda water']
+# letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# def letter (array,letter)
+#     final_array = []
+#     array.each do |word|
+#         if word.include?(letter)
+#             final_array << word
+#         end
+#     end
+#     final_array
+# end
+
+# p letter beverages_array,letter_o >> ['coffee', 'soda water']
+
+# p letter beverages_array, letter_t >> ['tea', 'water', 'soda water']
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
-# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
+# Process
+# Create a method called "new_array" that takes in a hash as a parameter
+# Retrieve the values from the hash using the hash.vales method
+# Get ride of the nested array using the flatten method
+# Alphabetize the values inside the array using the sort method
+# Return the "final_array"
+
+# us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+# # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
+
+# def new_array (hash)
+#     final_array = hash.values.flatten.sort
+#     final_array
+# end
```

Comment:
Nice job chaining those methods! Whenever you find yourself making a variable just to return it right afterwards, you can always just return the data. In this case, all that is needed in the `new_array` method is `hash.values.flatten.sort`!